uuid_test.go

```diff
@@ -874,20 +874,30 @@ func TestVersion7_pooled(t *testing.T) {
 func TestVersion7FromReader(t *testing.T) {
 	myString := "8059ddhdle77cb52"
 	r := bytes.NewReader([]byte(myString))
-	r2 := bytes.NewReader([]byte(myString))
-	uuid1, err := NewV7FromReader(r)
+	_, err := NewV7FromReader(r)
 	if err != nil {
 		t.Errorf("failed generating UUID from a reader")
 	}
 	_, err = NewV7FromReader(r)
 	if err == nil {
 		t.Errorf("expecting an error as reader has no more bytes. Got uuid. NewV7FromReader may not be using the provided reader")
 	}
-	uuid3, err := NewV7FromReader(r2)
-	if err != nil {
-		t.Errorf("failed generating UUID from a reader")
+}
+
+func TestVersion7Monotonicity(t *testing.T) {
+	length := 1000
+
+	uuids := make([]string, length)
+	for i := 0; i < length; i++ {
+		uuidString, _ := NewV7()
+		uuids[i] = uuidString.String()
+		time.Sleep(time.Millisecond)
```

Comment:
This forces this single test to take a full second.  A better test might be to force timeNow to return the same time on every call.  You don't need to store all the UUIDs, just the last one:

```
u, _ := NewV7()
lastUUID := u.String()
for i := 0; i < length; i++ {
        u, _ := Newv7()
        nextUUID := u.String()   
        if lastUUID >= nextUUID {
                t.Errorf("monotonicity failed at #%d: %s !< %s", i, lastUUID, nextUUID)
                break
        }
}
```

You could also run this test again where timeNow returns time incrementing by say 100 nanoseconds per call and if you want to go crazy, where timeNow returns time going backwards.